examples/guestbook/README.md

```diff
@@ -67,7 +67,82 @@ The web front end interacts with the redis master via javascript redis API calls
 
 This example requires a running Kubernetes cluster.  See the [Getting Started guides](../../docs/getting-started-guides/) for how to get started. As noted above, if you have a Google Container Engine cluster set up, go [here](https://cloud.google.com/container-engine/docs/tutorials/guestbook) instead.
 
-### Step One: Start up the redis master
+### Step One: Fire up the redis master service
+
+A Kubernetes [service](../../docs/user-guide/services.md) is a named load balancer that proxies traffic to one or more containers. This is done using the [labels](../../docs/user-guide/labels.md) metadata that we defined in the `redis-master` pod above.  As mentioned, we have only one redis master, but we nevertheless want to create a service for it.  Why?  Because it gives us a deterministic way to route to the single master using an elastic IP.
+
+Services find the pods to load balance based on the pods' labels.
+The selector field of the service description determines which pods will receive the traffic sent to the service, and the `port` and `targetPort` information defines what port the service proxy will run at.
+
+The file `examples/guestbook/redis-master-service.yaml` defines the redis master service:
+
+<!-- BEGIN MUNGE: EXAMPLE redis-master-service.yaml -->
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-master
+  labels:
+    app: redis
+    tier: backend
+    role: master
+spec:
+  ports:
+    # the port that this service should serve on
+  - port: 6379
+    targetPort: 6379
+  selector:
+    app: redis
+    tier: backend
+    role: master
+```
+
+[Download example](redis-master-service.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE redis-master-service.yaml -->
+
+Create the service by running:
+
+```console
+$ kubectl create -f examples/guestbook/redis-master-service.yaml
+service "redis-master" created
+```
+
+Then check the list of services, which should include the redis-master:
+
+```console
+$ kubectl get services
+NAME              CLUSTER_IP       EXTERNAL_IP       PORT(S)       SELECTOR                            AGE
+redis-master      10.0.136.3       <none>            6379/TCP      app=redis,role=master,tier=backend  1h
+```
+
+This will cause all pods to see the redis master apparently running on <ip>:6379.  A service can map an incoming port to any `targetPort` in the backend pod.  Once created, the service proxy on each node is configured to set up a proxy on the specified port (in this case port 6379).
+
+`targetPort` will default to `port` if it is omitted in the configuration. For simplicity's sake, we omit it in the following configurations.
+
+The traffic flow from slaves to masters can be described in two steps, like so:
+
+  - A *redis slave* will connect to "port" on the *redis master service*
+  - Traffic will be forwarded from the service "port" (on the service node) to the  *targetPort* on the pod that the service listens to.
```

Comment:
Near/at the end of each section that introduces a new concept/feature, please refer to the corresponding section of the user guide, in this case https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/connecting-applications.md